storage/remote/wal_watcher.go

```diff
@@ -220,9 +216,36 @@ func (w *WALWatcher) run() error {
 
 // findSegmentForIndex finds the first segment greater than or equal to index.
 func (w *WALWatcher) findSegmentForIndex(index int) (int, error) {
+	refs, err := w.segments()
+	if err != nil {
+		return -1, nil
+	}
+
+	for _, r := range refs {
+		if r >= index {
+			return r, nil
+		}
+	}
+
+	return -1, errors.New("failed to find segment for index")
+}
+
+func (w *WALWatcher) firstAndLast() (int, int, error) {
+	refs, err := w.segments()
+	if err != nil {
+		return -1, -1, nil
+	}
+
+	if len(refs) == 0 {
+		return -1, -1, nil
+	}
+	return refs[0], refs[len(refs)-1], nil
+}
+
+func (w *WALWatcher) segments() ([]int, error) {
```

Comment:
do we want to leave a comment clarifying that these are copied from tsdb/wal/wal.go` for now so that we don't need to open a `WAL`, and that we plan to move WAL watcher to tsdb?